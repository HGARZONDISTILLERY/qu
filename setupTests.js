import { QueryClient, QueryClientProvider } from 'react-query';
import { render } from '@testing-library/react';

const queryClient = new QueryClient();

const customRender = (ui, options) =>
  render(ui, { wrapper: ({ children }) => <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>, ...options });

export * from '@testing-library/react';
export { customRender as render };
