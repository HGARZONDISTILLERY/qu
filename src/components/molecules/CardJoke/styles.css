.joke-list-content {
  height: 100%;
  background-repeat: no-repeat;
  background-size: contain;
  background-position: 100px;
  position: relative
}

.joke-list-content h5 {
  background: yellow;
  padding: 5px;
  font-size: 16px;
}

.joke-list-content .MuiCardContent-root {
  text-align: left;
  max-width: 40%;
}

p.joke-list-votes {
  position: absolute;
  bottom: 5px;
  font-size: 12px;
}