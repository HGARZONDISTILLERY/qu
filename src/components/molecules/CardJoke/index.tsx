import React from 'react';
import "./styles.css";

import { Joke } from '../../../utils/types';

import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { Box } from '@mui/material';
import Meme1 from '../../../assets/meme-1.jpg';

interface CardJokeProps {
  joke: Joke;
}

const JokeListItem: React.FC<CardJokeProps> = ({ joke }) => {
  const card = (
    <Box className="joke-list-content" sx={{backgroundImage: `url('${Meme1}')`}}>
      <CardContent>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          {joke.setup}
        </Typography>
        <Typography variant="h5">
        {joke.punchline}
        </Typography>
      </CardContent>
    </Box>
  );

  return (
    <Card sx={{ minWidth: 275, maxWidth: 500 }} variant="outlined">{card}</Card>
  );
};

export default JokeListItem;