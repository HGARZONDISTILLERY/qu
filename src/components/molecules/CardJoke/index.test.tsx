import CardJoke from ".";
import { render } from "../../../../setupTests";

test("CardJoke component renders correctly", () => {
  const joke = {
    type: "programming",
    setup: "Why do C# and Java developers keep breaking their keyboards?",
    punchline: "Because they use a strongly typed language.",
    id: 74,
  };

  const tree = render(<CardJoke joke={joke} />);
  expect(tree).toMatchSnapshot();
});
