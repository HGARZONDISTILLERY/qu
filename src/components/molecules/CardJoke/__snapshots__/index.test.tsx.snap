// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardJoke component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1bbq0xd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="joke-list-content MuiBox-root css-s6ydel"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
            >
              Why do C# and Java developers keep breaking their keyboards?
            </p>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Because they use a strongly typed language.
            </h5>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1bbq0xd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="joke-list-content MuiBox-root css-s6ydel"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
          >
            Why do C# and Java developers keep breaking their keyboards?
          </p>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Because they use a strongly typed language.
          </h5>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
